### PR TITLE
Allow manual control of TooltipTrigger visibility

### LINKED ISSUE #536: Proposed Tooltip changes
Currently we're using the Tooltips, and the TooltipTrigger meets our needs better than the OverlayTrigger. However, we need the ability to manually control visibility of the tooltip programmatically, like the OverlayTrigger.

If I make a PR adding that, is that something you'd be willing to accept?

### PR DESCRIPTION
**Pull request by ahuth**
Resolves #536.

Adds the ability to manually control the visibility of Tooltips used with the TooltipTrigger, bringing it inline with the functionality of the OverlayTrigger.

### CODE REVIEW

**Comment by ahuth** [comment]
@charleshansen, how does this look to you, or is there anyone else I should hit up about this?

**Comment by charleshansen** [comment]
I would prefer if you made the behavior more similar to overlay_trigger. With overlay trigger, the `display` property is mostly respected even when trigger is set to click. This is handled in `componentWillReceiveProps` https://github.com/pivotal-cf/pivotal-ui/blob/master/src/react/overlay-trigger/overlay-trigger.js#L52-L54  

The net result is that if you hovering or clicking will toggle the display, but if you actually change the `display` prop yourself, that will also change the display.

In overlay trigger, the main effect of setting `trigger: manual` is to turn off the click and hover listeners. It doesn't do anything special to make the trigger start listening to `display: true`

**Comment by ahuth** [comment]
Alright, @charleshansen thanks for the feedback. The requested change has been made.

**Comment by charleshansen** [comment]
The behavior is a little better, but still doesn't actually capture the overlay trigger behavior, sorry it is a little tricky to explain.

The intended behavior is:
- On first render, respect the `display` prop
- If hover or click triggers a change, change visibility, and ignoring `display`
- When `display` changes, update visibility, ignoring whatever click or hover thinks
- If hover or click triggers another change,  update visibility, ignoring `display`

Basically, whenever either `display` or the trigger change, respect whichever changed most recently. The only way I know to do this is with `componentWillReceiveProps`.

Also, I'm going to pull in @jberney since he is the one who should ultimately merge the PR

**Comment by ahuth** [comment]
Okay, I think I got what we're trying to accomplish here. Thanks for the feedback

**Comment by ahuth** [comment]
@charleshansen @jberney is this more what you had in mind?

**Comment by ahuth** [comment]
Thanks!
